flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/RawToBinaryCastRule.java

```diff
@@ -0,0 +1,102 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.functions.casting;
+
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.LogicalTypeFamily;
+import org.apache.flink.table.types.logical.LogicalTypeRoot;
+import org.apache.flink.table.types.logical.utils.LogicalTypeChecks;
+
+import java.util.Arrays;
+
+import static org.apache.flink.table.codesplit.CodeSplitUtil.newName;
+import static org.apache.flink.table.planner.functions.casting.CastRuleUtils.accessField;
+import static org.apache.flink.table.planner.functions.casting.CastRuleUtils.methodCall;
+import static org.apache.flink.table.planner.functions.casting.CastRuleUtils.staticCall;
+
+/** {@link LogicalTypeRoot#RAW} to {@link LogicalTypeFamily#BINARY_STRING} cast rule. */
+public class RawToBinaryCastRule extends AbstractNullAwareCodeGeneratorCastRule<Object, byte[]> {
+
+    static final RawToBinaryCastRule INSTANCE = new RawToBinaryCastRule();
+
+    private RawToBinaryCastRule() {
+        super(
+                CastRulePredicate.builder()
+                        .input(LogicalTypeRoot.RAW)
+                        .target(LogicalTypeFamily.BINARY_STRING)
+                        .build());
+    }
+
+    /* Example generated code for BINARY(3):
+
+    byte[] deserializedByteArray$0 = result$2.toBytes(typeSerializer$5);
+    if (deserializedByteArray$0 != null) {
+        if (3 >= deserializedByteArray$0.length) {
+            result$4 = deserializedByteArray$0;
+        } else {
+            result$4 = java.util.Arrays.copyOfRange(deserializedByteArray$0, 0, 3);
+        }
+    } else {
+        result$4 = null;
+    }
+
+    */
+
+    @Override
+    protected String generateCodeBlockInternal(
+            CodeGeneratorCastRule.Context context,
+            String inputTerm,
+            String returnVariable,
+            LogicalType inputLogicalType,
+            LogicalType targetLogicalType) {
+        // Get length of target
+        final int targetLength = LogicalTypeChecks.getLength(targetLogicalType);
+
+        // Get serializer for RAW type
+        final String typeSerializer = context.declareTypeSerializer(inputLogicalType);
+        final String deserializedByteArrayTerm = newName("deserializedByteArray");
+
+        return new CastRuleUtils.CodeWriter()
+                .declStmt(
+                        byte[].class,
+                        deserializedByteArrayTerm,
+                        methodCall(inputTerm, "toBytes", typeSerializer))
+                .ifStmt(
+                        deserializedByteArrayTerm + " != null",
+                        thenWriter ->
+                                thenWriter.ifStmt(
+                                        targetLength
+                                                + " >= "
+                                                + accessField(deserializedByteArrayTerm, "length"),
```

Comment:
Can you swap this condition as it's easier to read?

Reply:
But why swap? To me at least, it's easier to say, `if (condition) then <normal> else <doSomethingmore>`

Reply:
With swap I mean `accessField(deserializedByteArrayTerm, "length")  + " <= " + targetLength`, which IMO is more readable

Reply:
More than adding the method `CastRuleUtils#accessField`, I think it makes sense to just add a method like `CastRuleUtils#accessArrayLength`. Can you also check if in the other rules the array length is used and replace it with `CastRuleUtils#accessArrayLength`